contracts/swaps/SwapsUser.sol

```diff
@@ -99,7 +100,8 @@ contract SwapsUser is State, SwapsEvents, FeesHelper {
 		uint256[3] memory vals,
 		bytes32 loanId,
 		bool miscBool, /// bypassFee
-		bytes memory loanDataBytes
+		bytes memory loanDataBytes,
+		bool isSwapExternal
```

Comment:
adding extra flag isn't elegant. this functions slowly grows into if-else hell, consider refactoring it in the future